frame/nomination-pools/src/lib.rs

```diff
@@ -582,25 +582,207 @@ pub struct PoolRoles<AccountId> {
+/// initial commission `current` value - the commission percentage and payee of commission.
+///
+/// The `max` commission value can only be decreased after the initial value is
+/// set, to prevent commission from repeatedly increasing.
+///
+/// An optional commission throttle allows the pool to set strict limits to how much
+/// commission can change in each update, and how often updates can take place.
+/// If a `throttle` is set before a commission `current` is set, the initial commission `current`
+/// value will not be subject to throttling. Subsequent commission updates will be.
+#[derive(
+	Encode, Decode, DefaultNoBound, MaxEncodedLen, TypeInfo, DebugNoBound, PartialEq, Copy, Clone,
+)]
+#[codec(mel_bound(T: Config))]
+#[scale_info(skip_type_params(T))]
+pub struct Commission<T: Config> {
+	/// Optional commission rate of the pool along with the account commission is paid to.
+	pub current: Option<(Perbill, T::AccountId)>,
+	/// Optional maximum commission that can be set by the pool `root`. Once set, this value
+	/// can only be updated to a decreased value.
+	pub max: Option<Perbill>,
+	/// Optional configuration around how often commission can be updated, and when the last
+	/// commission update took place.
+	pub throttle: Option<CommissionThrottle<T>>,
+}
+
+impl<T: Config> Commission<T> {
+	/// Get the current commission percentage of this pool.
+	///
+	/// Returns zero if commission has not yet been configured.
+	fn commission_or_zero(&self) -> Perbill {
+		self.current.as_ref().map(|(x, _)| *x).unwrap_or(Perbill::zero())
+	}
+
+	/// Returns true if a commission percentage updating to `to` would exhaust the throttle limit.
+	///
+	/// A commission update will be throttled (disallowed) if:
+	/// 1. not enough blocks have passed since the previous commission update took place, and
+	/// 2. the new commission is larger than the maximum allowed increase.
+	///
+	/// Throttling is not applied to commission updates if `current` is still `None`.
+	fn throttling(&self, to: &Perbill) -> bool {
+		if let Some(t) = self.throttle.as_ref() {
+			// check for `min_delay` throttling
+			if t.previous_set_at.map_or(false, |p| {
+				<frame_system::Pallet<T>>::block_number().saturating_sub(p) <
+					t.change_rate.min_delay
+			}) {
+				return true
+			}
+			// check for `max_increase` throttling
+			return (*to).saturating_sub(self.commission_or_zero()) > t.change_rate.max_increase
+		}
+		false
+	}
+
+	/// Set the pool's commission.
+	///
+	/// Update commission accordingly based on `commission` and `payee`. If
+	/// throttle is present, record the current block as the previously updated
+	/// commission. If the supplied commission is zero, `None` will be inserted
+	/// and `payee` will be ignored.
+	fn maybe_update_current(
+		&mut self,
+		commission: &Perbill,
+		payee: T::AccountId,
+	) -> DispatchResult {
+		ensure!(!(self.current.is_some() && self.throttling(&commission)), Error::<T>::CommissionChangeThrottled);
```

Comment:
Do we want to start throttling once `current` is set, as is in place now?

The reason for this is if we have a scenario like: set an initial `current` of `5%`, and then only allow an increase of `1%` every week (or slower, month). If `throttle` is set before `current`, it will now take 5 weeks or months to get to that desired 5%.

So this logic prevents this unfortunate mistake from the pool operator side.

Reply:
Upon revisiting, I think it is the operator's responsibility to not land into this scenario, and the UI can warn the operator if they attempt to set a throttle and commission has not yet been set. `current.is_some()` removed.